File: app/src/main/java/com/anadi/weatherinfo/repository/OpenWeatherMapInterface.java
package com.anadi.weatherinfo.repository;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface OpenWeatherMapInterface {
//    @GET("weather/")
//    void getWeather(@Query("q") String location,
//                    @Query("appid") String appid,
//                    @Query("units") String units,
//                    Callback<WeatherInfoSuper> cb);

    @GET("weather/")
    Call<WeatherInfo> getWeather(@Query("q") String location,
                                 @Query("appid") String appid,
                                 @Query("units") String units,
                                 @Query("lang") String lang);
}

File: app/src/main/java/com/anadi/weatherinfo/repository/data/OpenWeatherMapInterface.kt
package com.anadi.weatherinfo.repository.data

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface OpenWeatherMapInterface {
    //    @GET("weather/")
    //    void getWeather(@Query("q") String location,
    //                    @Query("appid") String appid,
    //                    @Query("units") String units,
    //                    Callback<WeatherInfoSuper> cb);
    @GET("weather/")
    fun getWeather(@Query("q") location: String?, @Query("appid") appid: String?,
                   @Query("units") units: String?, @Query("lang") lang: String?): Call<WeatherInfo?>?
}